packages/data-models/src/homonym.js

```diff
@@ -72,21 +72,29 @@ class Homonym {
   }
 
   static readObject (jsonObject) {
+    if (!jsonObject.form && !jsonObject.targetWord) {
+      throw new Error(Homonym.errMsgs.NO_TARGET_WORD_IN_JSON)
+    }
+    const targetWord = jsonObject.form || jsonObject.targetWord
     let lexemes = [] // eslint-disable-line prefer-const
     if (jsonObject.lexemes) {
       for (const lexeme of jsonObject.lexemes) {
         lexemes.push(Lexeme.readObject(lexeme))
       }
     } else {
-      const languageID = LMF.getLanguageIdFromCode(jsonObject.languageCode)
-      lexemes = [new Lexeme(new Lemma(jsonObject.targetWord, languageID), [])]
+      if (!jsonObject.languageCode) {
+        throw new Error(Homonym.errMsgs.NO_LANGUAGE_IN_JSON)
+      }
+      const lang = new Language(jsonObject.languageCode)
+      lexemes = [new Lexeme(new Lemma(targetWord, lang), [])]
     }
-    const homonym = new Homonym(lexemes, jsonObject.form || jsonObject.targetWord)
+    const homonym = new Homonym(lexemes, targetWord)
     homonym.lemmasList = jsonObject.lemmasList
     return homonym
   }
 
   convertToJSONObject (addMeaning = false) {
+    // TODO: Shall we use targetWord instead of form to match the internal homonym structure?
```

Comment:
we probably should. but let's create a separate issue to track that because it could impact the wordlist.

Reply:
I've created https://github.com/alpheios-project/alpheios-core/issues/596 to track this.